pages/survey.py

```diff
@@ -153,50 +153,68 @@ def finished():
 def video_submission():
     st.title("Video Upload")
 
-    # Initialize session state for tracking submission and upload progress
-    if "uploading" not in st.session_state:
-        st.session_state.uploading = False
+    st.markdown("Please submit your video through this [Google form](https://docs.google.com/forms/d/e/1FAIpQLSfDRHCootB91wKYUUvq5_qKmzk6lpYg0aS_adslML9dWkCCTQ/viewform).\n")
 
-    if "video_submitted" not in st.session_state:
-        st.session_state.video_submitted = False  # Tracks if the video was already submitted
+    st.write("After submitting the form, at the end there will be a password for you to enter below to complete the study.")
 
-    # File uploader that only accepts video files
-    uploaded_video = st.file_uploader("Upload a video file", type=["webm"])  # You can add other file types if needed
+    # Create a password input field
+    password = st.text_input("Enter password", type="password", key="password_video")
 
-    # Ensure the user uploads a video before enabling the submit button
-    if uploaded_video is None:
-        st.warning("Please upload a video file before proceeding.")
-        st.button("Submit", disabled=True)
-    else:
-        try:
-            st.success("Video uploaded successfully!")
-            # Display the video in the app
-            st.video(uploaded_video)
-
-            # Disable the submit button if the video has already been submitted
-            if st.button("Submit", key="submit_recording", disabled=st.session_state.uploading or st.session_state.video_submitted):
-                # Immediately disable the button to prevent spamming
-                st.session_state.uploading = True
-
-                # Progress bar
-                st.session_state.upload_progress = 0
-
-                # Call function to upload the large video to Google Drive
-                video_id = upload_to_drive(uploaded_video, uploaded_video.name)
-                st.success(f"Video uploaded successfully!")
-                print(f'video uploaded succesfully!  File ID: {video_id} for user: {st.session_state.username}')
-
-                # Update session state after submission to prevent future submissions
-                st.session_state.video_submitted = True
-                st.session_state.uploading = False
-
-                # Update user data and reset after completion
-                update_user_data("complete", 6)
-                st.session_state.last_progress = -1
-                st.rerun()  # Refresh the page to reflect changes
-        except Exception as e:
-            st.error(f"Error with the video save: {e}.\n Please contact for help.")
-            st.session_state.uploading = False
+    # Button to check the password
+    if st.button("Submit"):
+        if password == st.secrets["video_password"]['password']:
```

Comment:
nit: can we check password.strip() instead in case someone leaves a white space?